### ISSUE TITLE
Should TSLint run when there are syntactic errors in the file?

### ISSUE THREAD

**Issue by Xarkam** [palantir/tslint#3808]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.1
- __Running TSLint via__: Visual Studio

#### TypeScript code being linted

```ts
@NgModule({
  exports: [ RouterModule ],
  imports: [ RouterModule.forRoot(
    routes,
    { enableTracing: true },
  ],
})
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "no-duplicate-variable": true,
    "no-unused-variable": [
      true
    ],
    "linebreak-style":false,
    "max-line-length":false,
    "no-console":false,
    "no-shadowed-variable":false,
    "variable-name": [
      true,
      "check-format",
      "allow-leading-underscore",
      "allow-trailing-underscore"
    ]
  },
  "rulesDirectory": [
    "node_modules/tslint-eslint-rules/dist/rules"
  ]
}
```

#### Actual behavior
Because a picture is better than a long speech :
![image](https://user-images.githubusercontent.com/331816/38300957-88156f64-37fe-11e8-8da5-c43b07d249d1.png)
#### Expected behavior
Give good choice ;)

ps: yes, he misses one **)** in snippet

**Comment by grimly** [comment]
You have two errors at different places. Try validating this :

```ts
@NgModule({
  exports: [ RouterModule ],
  imports: [ RouterModule.forRoot(
    routes,
    { enableTracing: true, }
  )],
})
```

I switched `,` and `}` to add the trailing comma in the object literal and remove the one in the `forRoot` function call. 

And I added the `)` back in the end of the `forRoot` function call.

**Comment by Xarkam** [comment]
Yes, I have specified the missing one `)`
With the final `)` I've not error.

But it's disturbing to have this message with just a missing `)`

**Comment by grimly** [comment]
This is odd, but when it come to parsing, the missing closing parenthesis may have seen something completely different to what we may think.

It might have back-tracked to something that is not the `RouterModule.forRoot` function call.

I don't think the linter can do a lot here.

**Comment by JoshuaKGoldberg** [comment]
The root issue here is, as suggested, that TSLint should have a better strategy for providing failures when there's a syntax problem. Modifying the title to make that more clear; I must have rediscovered this issue a half dozen times by now. 😆

**Comment by JoshuaKGoldberg** [comment]
Per #3946: what about --fix?

**Comment by Xarkam** [comment]
Hello, I think TSLint can stop on misconfiguration.
But, it needs a clear message that TSLint is stopped because of a bad configuration.

Developers can then narrow their search field to tslint configuration on errors.

**Comment by MVrachev** [comment]
Hi, I just read through the discussion here and it's little unclear for me what decision was made.

Is there a way ESLint reports for syntax errors? I imagine that syntax errors will prevent the correct analysis of a file so it makes sense to report those problems.

The way I saw [Gosec](https://github.com/securego/gosec) (Golang static code analysis tool) handles the syntax errors is by giving Golang syntax errors in the beginning of the output.

Text: 
![image](https://user-images.githubusercontent.com/16246778/54884910-1f08e300-4e7f-11e9-936c-e92fc32306db.png)

JSON:
![image](https://user-images.githubusercontent.com/16246778/54884913-2fb95900-4e7f-11e9-9a3a-e972d151b895.png)

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
